com.unity.ml-agents/Runtime/InferenceBrain/ApplierImpl.cs

```diff
@@ -13,18 +13,21 @@ namespace MLAgents.InferenceBrain
     /// </summary>
     public class ContinuousActionOutputApplier : TensorApplier.IApplier
     {
-        public void Apply(TensorProxy tensorProxy, IEnumerable<AgentIdActionPair> actions)
+        public void Apply(TensorProxy tensorProxy, IEnumerable<int> actionIds, Dictionary<int, float[]> lastActions)
         {
             var actionSize = tensorProxy.shape[tensorProxy.shape.Length - 1];
             var agentIndex = 0;
-            foreach (var idActionPair in actions)
+            foreach (int agentId in actionIds)
             {
                 var actionValue = new float[actionSize];
                 for (var j = 0; j < actionSize; j++)
                 {
                     actionValue[j] = tensorProxy.data[agentIndex, j];
                 }
-                idActionPair.action.Invoke(new AgentAction { vectorActions = actionValue });
+                if (lastActions.ContainsKey(agentId))
```

Comment:
Memory reuse optimization part 2 - if `lastActions.ContainsKey(agentId) && lastActions[agentId].Length == actionSize` then reuse the value there and don't `new` at the start.